src/main/java/bio/terra/service/dataset/flight/ingest/IngestPopulateFileStateFromFlightMapStep.java

```diff
@@ -66,6 +80,31 @@ public StepResult doSkippableStep(FlightContext context) {
     return StepResult.getStepResultSuccess();
   }
 
+  /**
+   * Filter the stream so that only files that need to be ingested are ingested, but also save the
+   * files that have already been ingested.
+   *
+   * @param bulkLoadModel A candidate file to maybe ingest
+   * @param collector A place to put the FileModels that already exist
+   * @return A Stream of BulkLoadFileModel if it needs to be ingested, or an empty Stream if not.
+   */
+  private Stream<BulkLoadFileModel> fileNotIngested(
+      BulkLoadFileModel bulkLoadModel, List<FileModel> collector) {
+    int depth = bulkLoadModel.getTargetPath().replaceAll("[^/]*", "").length();
```

Comment:
I'm all for avoiding 3rd party APIs but this seems worse to me (harder to read, slower to execute) than calling `StringUtils.countMatches()`